drivers/peci/peci_mchp_xec.c

```diff
@@ -135,7 +136,13 @@ static void peci_xec_bus_recovery(const struct device *dev, bool full_reset)
 	LOG_WRN("%s full_reset:%d", __func__, full_reset);
 	if (full_reset) {
 		base->CONTROL = MCHP_PECI_CTRL_PD | MCHP_PECI_CTRL_RST;
-		k_busy_wait(PECI_RESET_DELAY);
+
+		if (k_is_in_isr()) {
+			k_msleep(PECI_RESET_DELAY_MS);
+		} else {
+			k_busy_wait(PECI_RESET_DELAY);
+		}
```

Comment:
`k_msleep()` should not be used in interrupt context. So the if-else blocks should be reversed.

Reply:
Corrected